app/models/user.rb

```diff
@@ -65,12 +65,12 @@ def passed_assignments
     assignments.where(status: Status::Passed.to_i)
   end
 
-  def comments
-    assignments.flat_map(&:comments)
+  def messages
+    Message.where(submission_id: assignments.pluck(:submission_id)).order('created_at DESC')
```

Comment:
If `assignment` were a normal association in `Messages`, we could just do a simple `has_many through:`.